llvm/lib/Target/AIE/AIE2InstrPatterns.td

```diff
@@ -779,7 +779,9 @@ def : VEQZ_INSTR<int_aie2_veqz32, VEQZ_32>;
 
 // VBCST
 def : Pat<(int_aie2_vbroadcast8_I512 eR:$s0),  (VBCST_8 eR:$s0)>;
+def : Pat<(int_aie2_vbroadcast8_I512 (and eR:$rs1, 255)), (VBCST_8 eR:$rs1)>;
 def : Pat<(int_aie2_vbroadcast16_I512 eR:$s0), (VBCST_16 eR:$s0)>;
+def : Pat<(int_aie2_vbroadcast16_I512 (and eR:$rs1, 65535)), (VBCST_16 eR:$rs1)>;
```

Comment:
Do we still need the DemandedBits then? It does not bother me to keep the `zext`, but I get @martien-de-jong and @konstantinschwarz preference for keeping the semantics visible.

Reply:
Strictly speaking no, but cleaner IR might unlock other optimizations.
However these patterns only cover zero extensions. Maybe add patterns for sign-extensions as well?